main.py

```diff
@@ -0,0 +1,420 @@
+        # Seek to the end of the file and get its length
+        SEEK_FROM_END = 2
+        SEEK_FROM_BEG = 0
+        f.seek(0, SEEK_FROM_END)
+        total_size = f.tell()
+
+        # Determine the requsted range of content
+        request_range = self.headers.get('Range')
+        if request_range:
+            self.log_message('request range: %s', request_range)
+
+            # Parse range units per https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Range#syntax
+            range_unit = 'bytes'
+            if '=' in request_range:
+                range_unit, request_range = request_range.split('=')
+
+            # Only byte ranges are supported, other units cannot be processed
+            if range_unit != 'bytes':
+                self.log_message('unsupported range: non-bytes unit was specified')
+                return self.send_error(HTTPStatus.REQUESTED_RANGE_NOT_SATISFIABLE)
+
+            # Do not support multiple byte range requests
+            if ',' in request_range:
+                self.log_message('unsupported range: multiple ranges were specified')
+                return self.send_error(HTTPStatus.REQUESTED_RANGE_NOT_SATISFIABLE)
+
+            # Process range start/end per https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Range#directives
+            range_start, range_end = request_range.strip().split('-')
+
+            # Try to convert range start/end to byte counts
+            try:
+                self.log_message('range_start: %s', range_start)
+                if range_start:
+                    range_start = int(range_start)
+                self.log_message('range_end: %s', range_end)
+                if range_end:
+                    range_end = int(range_end)
+            except Exception:
+                self.log_message('unsupported range: failed to convert ranges to integers')
+                return self.send_error(HTTPStatus.REQUESTED_RANGE_NOT_SATISFIABLE)
+
+            # Require range start positions
+            if range_start == '':
+                self.log_message('unsupported range: missing range-start')
+                return self.send_error(HTTPStatus.REQUESTED_RANGE_NOT_SATISFIABLE)
+
+            # Do not support range end
+            if range_end != '':
+                self.log_message('unsupported range: range-end was specified')
+                return self.send_error(HTTPStatus.REQUESTED_RANGE_NOT_SATISFIABLE)
+
+            # Seek to the beginning of the range in the content
+            f.seek(range_start, SEEK_FROM_BEG)
+
+            # Set the approtriate code/headers for ranged content
+            self.send_response(HTTPStatus.PARTIAL_CONTENT)
+            self.send_header('Content-Range', f'bytes {range_start}-{total_size-1}/{total_size}')
+            self.send_header('Content-Length', str(total_size-range_start))
+        else:
+            # Seek to the beginning of the content since no range was specifieg
```

Comment:
Could you clarify this comment? There is a typo.

Reply:
https://github.com/duckflix/media-server/issues/5